sixtools/AD_Handler.py

```diff
@@ -0,0 +1,68 @@
+from databroker.assets.handlers_base import HandlerBase
+import os.path
+
+
+class HDF5SingleHandler(HandlerBase):
+    '''Handler for hdf5 data stored 1 image per file.
+    Parameters
+    ----------
+    fpath : string
+        filepath
+    template : string
+        filename template string.
+    filename : string
+        filename
+    key : string
+        the 'path' inside the file to the data set.
+    frame_per_point : float
+        the number of frames per point.
+    '''
+    specs = {'AD_HDF5_SINGLE'} | HandlerBase.specs
+
+    def __init__(self, fpath, template, filename, key, frame_per_point=1):
+        self._path = os.path.join(fpath, '')
+        self._fpp = frame_per_point
+        self._template = template
+        self._filename = filename
+        self._key = key
+
+    def _fnames_for_point(self, point_number):
+        start = int(point_number * self._fpp)
+        stop = int((point_number + 1) * self._fpp)
+        for j in range(start, stop):
+            yield self._template % (self._path, self._filename, j)
+
+    def __call__(self, point_number):
+        ret = []
+        import h5py
+        for fn in self._fnames_for_point(point_number):
+            f = h5py.File(fn, 'r')
+            data = f[self._key]
```

Comment:
Contrary @stuwilkins I think this should be returning numpy arrays so we do not end up with leaking file handles or many instances of the same file open at the same time.

the `h5py.Dataset` objects look enough like numpy arrays that things will work, but have have very different performance issue.  Repeatedly slicing into a numpy array is very cheap (it just does some pointer work and returns a view in most cases), on the other hand slicing into a Dataset actually goes and gets the bits off disk.

Reply:
I do think that this should stay the business of the handlers and the Header should stay out of it.  Cases where I think returning datasets makes sense is if we end up with, for example, a "Sinogram_stream" in tomography which is a 1 event stream which gives you a hook to the 3D sinogram, but I do not think that the current case at SIX has the size issues that would push us that direction yet.

Given that we know these files are feature positions, what about the idea of returning dataframes here?

Reply:
likewise to the above statement, can we transfer it to the databroker PR (here)[https://github.com/NSLS-II/databroker/pull/375] instead of assesing it at this point.

This would need to be part of a larger discussion, I am against making this a 'special' case so if we want to make that change here (which I think we probably do) we should also make it for all of the hdf5 handlers.

Reply:
The interesting thing here is not the file format, but the semantics of the data inside of the file.  This really is a table, the earlier we can express it as that the better.

Reply:
Right, since we want to switch from sixtools to databroker to be transparent to users, we should nail down the return type in this PR.

All the other HDF5 handlers are returning image data. This is returning a table of numbers with variable length. Representing that as a DataFrame will likely make the downstream analysis code prettier, don't you think?

Reply:
As discussed in person and over skype, we will update this version to be bespoke to the rixscam detector data output type. A second handler will be added to databroker that will work with all hdf5_plugins operating in `Single` mode and will return the same datatype as that for `Streaming` mode.